tactics/class_tactics.ml

```diff
@@ -525,7 +525,8 @@ let top_sort evm undefs =
   let l' = ref [] in
   let tosee = ref undefs in
   let rec visit ev evi =
-    let evs = Evarutil.undefined_evars_of_evar_info evm evi in
+    (* NB: caching of undefined evars is discarded here *)
+    let (evs,_) = Evarutil.undefined_evars_of_evar_info ev evm evi in
```

Comment:
This code wasn't on the slow path we had found, so I made the minimal change I could. Also, this function doesn't return an evar map to its callers, so the caches won't get propagated.

But I can propagate the caches within the loops inside this function, so I'll try that.